### PR TITLE
Document message validation option accepts Proc [ci skip]

### PR DESCRIPTION
**Pull request by eliotsykes**

### CODE REVIEW

File: guides/source/active_record_validations.md [hunk 1]
```diff
@@ -777,7 +777,21 @@ Topic.create(title: nil).valid? # => true
...
-for each validation helper.
+for each validation helper. The `:message` option accepts a `String` or `Proc`.
+
+```ruby
+class Person < ActiveRecord::Base
+  # Hard-coded message
+  validates :name, presence: { message: "must be given please" }
+
+  # Message with dynamic attribute value
+  validates :age, numericality: { message: "%{value} seems wrong" }
```
Comment (schneems) [inline guides/source/active_record_validations.md]: Is value a variable that is auto populated? That's what this example makes it look like, if that's the case we should explicitly say that.

File: guides/source/active_record_validations.md [hunk 2]
```diff
@@ -777,7 +777,21 @@ Topic.create(title: nil).valid? # => true
...
+
+  # Proc result as a message
+  validates :username,
+    uniqueness: { message: -> "taken! Try again #{Time.zone.tomorrow}" }
```
Comment (schneems) [inline guides/source/active_record_validations.md]: Don't we need brackets for the lambda? Also does the lambda get passed any values like the object being validated against? If so can you add another example to show that?

**Comment by eliotsykes** [comment]
Thanks for the much-needed feedback @schneems!

**Comment by senny** [comment]
@eliotsykes thank you :yellow_heart:

**Pull request merged**